resources/analysisTools/rnaseqworkflow/navBashLib.sh

```diff
@@ -10,6 +10,10 @@ function echo_run (){
         if [[ "$TEST_RUN" != true ]]
         then
                 eval $COMMAND
+                if [[ $? -ne 0 ]]
```

Comment:
Instead of checking the job exit code, we checked only the presence of result files. In these scenarios, the workflow continued after a session crash since there was an empty result file. Not sure if this was intentional behaviour.

Reply:
I also don't know. Could be that there were programs for which an exit-code != 0 is o.k., but for these actually a specific should be implemented, instead of a general ignore of the exit code. Apparently, `set -e` was also not set by default (in which case the current addition would be superfluous). You could ask @naveedishaque.

Reply:
You could save the exit code and then below do `exit $exitCode`.

Reply:
Kinda agree with Nags, but catching error codes is still relevant. I like the suggestion of `exit $exitCode`.
Also checking for the results file makes sense. Perhaps `check_or_die()` could be extended to include an optional 3rd variable which checks a minimum line or word count of the file to deal with the empty file case?

Reply:
I have now added the `$exitCode` to the `echo_run` function exit message. This should be good and lets the workflow fail if there is any non-zero exit.